### ISSUE TITLE
TCA 1.7.0 minimum version error iOS15

### ISSUE THREAD

**Issue by artemkrachulov** [pointfreeco/swift-composable-architecture#3033]
### Description

After the TCA update from version 1.6 to 1.7 got the error:

<img width="1020" alt="Screenshot 2024-04-30 at 2 39 57 PM" src="https://github.com/pointfreeco/swift-composable-architecture/assets/7997760/37627602-87bf-4e63-8a68-30ab8ac7e3e1">

The project has a minimum version iOS 15. 

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Steps to reproduce

1. Create a new project 
2. Lower ios version to ios15
3. Add TCA dependency with exact version 1.7.0
4. Build

### The Composable Architecture version information

1.7.0

### Destination operating system

iOS 15

### Xcode version information

Version 15.2 (15C500b)

### Swift Compiler version information

```shell
swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
Target: x86_64-apple-macosx14.0
```

**Comment by artemkrachulov** [comment]
Description says https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/observationbackport#Overview

> With version 1.7 of the Composable Architecture we have introduced support for Swift 5.9’s observation tools, and we have backported those tools to work in iOS 13 and later. Using the observation tools in pre-iOS 17 does require a few additional steps and there are some gotchas to be aware of.

As I understood it should work out of the box.

**Comment by valodyaios** [comment]
<img width="1102" alt="Screenshot 2024-04-30 at 17 15 08" src="https://github.com/pointfreeco/swift-composable-architecture/assets/98346572/c799f868-0610-4b70-8ad7-34e117e086e1">

I have the same problem. I have been using 1.7.0 version so far, but starting from yesterday the project is no longer possible to build.

**Comment by alejandro-honor-ed** [comment]
Having the exact same issue at the moment and it's critical for us to figure it out since our CI and release train is currently on hold until this is figured out. I'm trying to figure out how to fix it / pin swift-perception to a specific version, but I'll update in a few minutes.

**Comment by alejandro-honor-ed** [comment]
This made the trick https://github.com/pointfreeco/swift-composable-architecture/issues/3028#issuecomment-2083573439. I'm on TCA 1.8.0 and I pinned swift-perception to 1.1.1 and that worked perfectly fine.

**Comment by valodyaios** [comment]
Thanks @alejandro-honor-ed it works for me (added https://github.com/pointfreeco/swift-perception with fixed 1.1.5)

**Comment by pm-dev** [comment]
Should be fixed now
https://github.com/pointfreeco/swift-perception/releases/tag/1.1.7

**Comment by mbrandonw** [comment]
This was also fixed version 1.10.1 of TCA, so you can fix by pinning to TCA 1.10.1 or swift-perception 1.1.7. Sorry for the churn! We tried fixing a problem that only affected people still on iOS 17.0.0 _beta_, and sadly it caused more problems than it solved.

**Comment by mbrandonw** [comment]
@pm-dev The "exact" option in SPM should really only be used close the root of your application, ideally only the app target, and sparingly even then. It's just that "exact" dependencies make it very easy to run into un-resolvable dependency graphs, and so should be used with care, and only temporarily. So we would never want to do it in TCA.
